apps/examples/e2e/tests/test-ui.spec.ts

```diff
@@ -1,10 +1,109 @@
-import test from '@playwright/test'
+import test, { expect } from '@playwright/test'
 import { setup } from '../shared-e2e'
 
-test.describe('ui', () => {
+test.describe('mobile ui', () => {
 	test.beforeEach(setup)
-
-	test('mobile style panel opens and closes when tapped', async ({ isMobile }) => {
+	test('mobile style panel opens and closes when tapped', async ({ isMobile, page }) => {
+		test.skip(!isMobile, 'only run on mobile')
+		expect(await page.isVisible('.tlui-style-panel')).toBe(false)
+		await page.getByTestId('mobile-styles.button').click()
+		expect(await page.isVisible('.tlui-style-panel')).toBe(true)
+		await page.getByTestId('mobile-styles.button').click()
+		expect(await page.isVisible('.tlui-style-panel')).toBe(false)
+	})
+	test('mobile style panel closes when tapping off the menu', async ({ isMobile, page }) => {
 		test.skip(!isMobile, 'only run on mobile')
+		await page.getByTestId('mobile-styles.button').click()
+		expect(await page.isVisible('.tlui-style-panel')).toBe(true)
+		await page.getByTestId('tools.select').click()
+		expect(await page.isVisible('.tlui-style-panel')).toBe(false)
+	})
+	test('mobile style menu button is disabled for the eraser tool', async ({ isMobile, page }) => {
+		test.skip(!isMobile, 'only run on mobile')
+		const eraserTool = page.getByTestId('tools.eraser')
+		await eraserTool.click()
+		const mobileStylesButton = page.getByTestId('mobile-styles.button')
+		expect(await mobileStylesButton.isDisabled()).toBe(true)
+	})
+})
+
+test.describe('when selecting a tool from the toolbar', () => {
+	test.beforeEach(setup)
+	test('selecting a tool changes the button color', async ({ page }) => {
+		const selectTool = page.getByTestId('tools.select')
+		const drawTool = page.getByTestId('tools.draw')
+		await selectTool.click()
+		// We can't check the style of pseudo elements, as the browser doesn't expose them to the test
+		// So we use text color as a proxy
+		expect(selectTool).toHaveCSS('color', 'rgb(255, 255, 255)')
```

Comment:
Maybe we can introduce helper functions like `toBeSelected` which do this assertion? Would make it more obvious what the exact intention is.